pkg/handler/aws/uploadProfilePicture/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/CollActionteam/collaction_backend/auth"
+	"github.com/CollActionteam/collaction_backend/internal/uploads"
+	awsRepository "github.com/CollActionteam/collaction_backend/pkg/repository/aws"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+func handler(ctx context.Context, req events.APIGatewayV2HTTPRequest) (res events.APIGatewayProxyResponse, err error) {
+
+	// This is to recover from a "invalid memory address or nil pointer dereference: errorString" runtime error
+	// when the endpoint is called without valid auth token
+	defer func() {
+		if r := recover(); r != nil {
+			res = events.APIGatewayProxyResponse{StatusCode: http.StatusInternalServerError}
+			err = fmt.Errorf(fmt.Sprintf("%v", r))
+			return
+		}
+	}()
+
+	usrInf, err := auth.ExtractUserInfo(req)
+	if err != nil {
+		res = events.APIGatewayProxyResponse{StatusCode: http.StatusForbidden, Body: "user not authorized"}
+		return res, err
+	}
+
+	// getting user id, which will be used as object key
```

Comment:
Noted, I left the comment just as it was in the code before but I would remove them

Reply:
I generally refactored the code the into the hexagonal structure without changing most of the logic since it was working like that before. The recover function isn't needed since err is being checked and sent to the user